src/grpc/packet_router/hpr_packet_router_service.erl

```diff
@@ -0,0 +1,196 @@
+-module(hpr_packet_router_service).
+
+-behaviour(helium_packet_router_packet_bhvr).
+
+-export([
+    init/2,
+    route/2,
+    handle_info/2
+]).
+
+-export([
+    send_packet_down/2,
+    locate/1,
+    register/1
+]).
+
+-define(REG_KEY(Gateway), {?MODULE, Gateway}).
+
+-spec init(atom(), StreamState :: grpcbox_stream:t()) -> grpcbox_stream:t().
+init(_RPC, StreamState) ->
+    StreamState.
+
+-spec route(hpr_envelope_up:envelope(), grpcbox_stream:t()) ->
+    {ok, grpcbox_stream:t()} | {stop, grpcbox_stream:t()}.
+route(EnvUp, StreamState) ->
+    case hpr_envelope_up:data(EnvUp) of
+        {packet, PacketUp} ->
+            _ = erlang:spawn(hpr_routing, handle_packet, [PacketUp]),
+            {ok, StreamState};
+        {register, Reg} ->
+            PubKeyBin = hpr_register:gateway(Reg),
```

Comment:
We could use the register message as a chance to add `lager:md/1` to the process.